packages/core/integration-tests/test/lazy-compile.js

```diff
@@ -261,4 +274,73 @@ describe('lazy compile', function () {
     ]);
     subscription.unsubscribe();
   });
+
+  it('should lazy compile properly when an assets needs to be re-visited in symbol propagation', async () => {
+    await fsFixture(overlayFS, __dirname)`
+      lazy-compile-import-symbols
+        index.js:
+          import { shared, async } from './main';
+          export default Promise.all([shared(), async()]);
+
+        main.js:
+          export function shared() {
+            return import('./shared');
+          }
+          export function async() {
+            return import('./async');
+          }
+
+        async.js:
+          // Trigger more deps so a full 'propagateSymbolsUp' pass is executed
```

Comment:
Do you know why this only happens if the full pass runs? Feels odd to me

Reply:
There are other circumstances that can cause this to occur outside of a full pass, however this was the easiest way to reflect in a test. The reason is fails though is because the full pass only applies to propagateSymbolsUp. However, in this case there are nodes that are not re-visited by propagateSymbolsDown and contain stale data.